nextflow_schema.json

```diff
@@ -96,8 +98,7 @@
                 },
                 "wfmash_exclude_delim": {
                     "type": "string",
-                    "description": "Skip mappings between sequences with the same name prefix before the given delimiter character. [default: all-vs-all and !self].",
-                    "hidden": true
+                    "description": "Skip mappings between sequences with the same name prefix before the given delimiter character. [default: all-vs-all and !self]."
```

Comment:
Not sure this is a good idea, because the default value is like `-X` and not want you want to place here as default. Although, I get the point.

Reply:
and `-X` stands for "all-vs-all and !self"?

Reply:
from the `wfmash` docs:

```
        -X, --skip-self                   skip self mappings when the query and target name is the
                                          same (for all-vs-all mode)
```

Reply:
per default, the pipeline has `-X` already set

Reply:
all-vs-all is implicit as we run

`wfmash input.fasta input.fasta -X ...`

Reply:
then I would not say `default`. If I understand you correctly, than this value is not a string but a boolean?

Reply:
ah, I am confusing parameters I think, let me read up again... `wfmash` has too many parameters

Reply:
Alright, here is what is going on: Per default we run `wfmash` in all-vs-all mode. This is just given by how we invoke it: `wfmash input.fa input.fa`. However, there can we 2 cases where we can achieve additional speedup:

1. We are never! interested in self-mappings (we would map the same sequence to itself again, but that won't does not give us any helpful information). Therefore, per default, the flag `-X` in  all `wfmash` invocations is set. Which is the current default, so --> `all-vs-all and !self`. Not sure how to express this better and where to put it.
2. We have as input several chromosomes, but they are still on the contig level. So we want to prevent mappings of sequences that originate from the same chromosomes. with `wfmash_exclude_delim` we specify a delimiter (usually `#`) that is in the name of each FASTA sequence by which we are able to determine which sequences belong to the same chromosome.

So yeah, not sure how to express the default here. The value of `wfmash_exclude_delim` itself is none. And this means we run the pipeline in `all-vs-all and !self` mode.

Reply:
The value of `wfmash_exclude_delim` in `nextflow.config` is indeed null. Shall I add this to the schema @mashehu ?

Reply:
it's fine, we don't (yet) record default values with `null` in the schema. the parameter definition is still unclear to me, because it reads like I can either skip or not skip with the same prefix... ah is the delim deliminating what is prefix and what is not?

Reply:
Yes, that's the idea! It is not yet mandatory, but in the future the pipeline will only accept FASTAs which sequence names follow the https://github.com/pangenome/PanSN-spec. Then some parameters can be automated even better.

Reply:
Alright, I pushed a more detailed explanation removing the weird default text.